### ISSUE TITLE
Couldn't detect NPE when calling static method

### ISSUE THREAD

**Issue by Koc** [kalessil/phpinspectionsea#689]
![default](https://user-images.githubusercontent.com/191082/33968261-c61dd9e0-e06f-11e7-830c-5fa5905e17ed.png)

expected warning about NPE when calling `convertDateTimeToDateKey`

**Comment by rentalhost** [comment]
Okay, I could reproduce it.

```php
<?php

declare(strict_types = 1);

class Example
{
    public function checkIssue(): void
    {
        // Don't warn. Fail. Both should be warned.
        $this->expectDateTime($this->returnNullable());
        $this->expectDateTime(null);

        // Warn okay (EA inspection).
        $dateTime = $this->returnNullable();
        $this->expectDateTime($dateTime);

        // Warn okay (PS inspection).
        $this->expectDateTime(new stdClass);
    }

    public function expectDateTime(DateTime $dateTime): void
    {
    }

    public function returnNullable(): ?DateTime
    {
        return new DateTime;
    }
}

```

**Comment by funivan** [comment]
@rentalhost I think this can go to the Ultimate ;)

**Comment by kalessil** [comment]
@Koc : any chance that you would like to migrate onto https://plugins.jetbrains.com/plugin/10215-php-inspections-ea-ultimate- (commercial and enhanced version of the plugin) ? I'm asking as I need to make a decision which of plugins getting this enhancement.

**Comment by Koc** [comment]
not now, but you can move to commercical version, no problem

**Comment by kalessil** [comment]
Will it be fair in your opinion?

**Comment by Koc** [comment]
100%. A agree that this plugin prevents errors in some cases and make my work much productive. Just for now I have some financing problems and maybe will buy plugin next year.

**Comment by kalessil** [comment]
Thank you!

**Comment by Koc** [comment]
@kalessil can you create comparation table for free/paid version and include link on it in plugin pages?

**Comment by kalessil** [comment]
Will do (next days if not get ill completely)

**Comment by kalessil** [comment]
Implemented, the comparison list to be published in early January.

**Comment by kalessil** [comment]
@Koc : here we go with Ultimate-plugin details https://kalessil.github.io/page-blog-post-about-ea-ultimate.html
